### DISCUSSION TITLE
Auslesen der Überschussleistung mit Python

### DISCUSSION THREAD

**Discussion by BossoH** [camueller/SmartApplianceEnabler#197] [Q&A] [answered]
Hallo,
gibt es eine Möglichkeit den Wert für die aktuell überschüssige, also eingespeiste Leistung mit Python auszulesen?

Hintergrund der Frage
Ich mache mit meiner Überschussenergie Wasser in einem Boiler warm. Die Leistung des Heizstabes steuere ich mit einem Kemo M240 Leistungsregler. Theoretisch ist damit eine stufenlose Regelung zwischen 0 und 2000W möglich. Ich muss aber für jede Stufe einen separaten Heizstab im SAE definieren und benötige für jede Stufe einen Relaiskontakt. Praktisch habe ich deshalb nur 4 Stufen. Außerdem kann ich den Verbrauch mit einem Zähler nicht vernünftig abbilden. 
Könnte ich die überschüssige Leistung via Python abfragen, würde ich das als einen programmgesteuerten Verbraucher anlegen und würde die Regelung über das Skript machen. 

Danke

Edit: Ich benutze Version 2.x

**Comment by camueller** [comment, marked as answer]
Das SEMP-Protokoll gibt das nicht her und der SAE hat diese Information nicht.

**Reply by BossoH** [reply]
Da habe ich wohl Pech gehabt. Trotzem danke.

**Comment by camueller** [comment]
Du kannst ja selbst Deinen Erzeugungszähler und Verbrauchszähler auslesen und die Differenz bilden. Ist halt ziemlich viel Bastelei ...

Grundsätzlich kann der Sunny Home Manager mit Geräten mit variabler Leistungsaufnahme umgehen. Dieses Feature kam im Rahmen der Unterstützung von Wallboxen dazu. Der SAE untersützt das auch, aber eben nur bei Wallboxen. Grundsätzlich könnte man auch den SAE erweitern, aber dafür wäre Einiges zu ändern. Vermutlich müsste man einer Appliance mehrere Schalter zuordnen können und es müsste Regeln geben, wie ein Einschalt-Befehl mit 1400W auf die Schalter verteilt wird. Für so viel Aufwand ist das Interesse an einem solchen Feature nicht gross genug.

**Comment by BossoH** [comment]
Der Erzeugerzähler ist der Wechselrichter selbst, der "spricht" Modbus. Aber der Verbrauchszähler ist in meinem Fall nur der Sunny Home Manger 2. Damit funktioniert deine Idee soweit ich das verstanden habe nicht. 
So wie es ist funktioniert es ja. Ich kann damit gut leben.

**Reply by m16dd** [reply]
Hallo,
hat jemand Erfahrung mit einem DRT428m? Ich versuche vergeblich das Teil über modbus auszulesen. Der SDM 120 lässt sich leicht ansprechen. Kann mir hier jemand weiterhelfen?

**Reply by camueller** [reply]
Du hat `mbusd` laufen und Deinen Modbus an beiden Enden terminiert? Eigentlich sollte es mit den Zählern von b+g eTech bzw. eastron keine Probleme geben. Habe selbst einen SDM 220 und einen SDM 72D-M per Modbus angebunden.
Für weitere Diskussionen zu dem Thema bitte neuen Thread aufmachen.
